src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle.TypeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.CSharp.Utilities
+{
+    internal sealed class CSharpUseImplicitTypeHelper : CSharpTypeStyleHelper
```

Comment:
This was moved down from CSharpFeatures. Other than renaming it, and making it a singleton, it was not changed.